lib/dialects/sqlite3/schema/sqlite-tablecompiler.js

```diff
@@ -266,8 +266,16 @@ class TableCompiler_SQLite3 extends TableCompiler {
   }
 
   _setNullableState(column, isNullable) {
-    const fnCalled = isNullable ? '.setNullable' : '.dropNullable';
-    throw new Error(`${fnCalled} is not supported for SQLite.`);
+    const compiler = this;
+
+    this.pushQuery({
+      sql: `PRAGMA table_info(${this.tableName()})`,
```

Comment:
We've been aiming to make SQLite support in knex more friendly towards use-cases where one might want to use knex as a pure query builder and just generate SQL; to that end it is preferable to not have this kind of pseudo-sql.

`foreign` is a better example of multi-statement operation, see how it utilizes `statementsProducer`

Reply:
Sure, I can look into that. As mentioned, the implementation was based on the `alterColumn` functions which seemed more analogous to what I was trying to achieve. 👍

Reply:
@kibertoad Does all look correct to you now?

Reply:
Could you also add an assertion to your test, similar to the one here https://github.com/knex/knex/blob/a75d0194cf42dceb95019ea22c6a289afb1f3397/test/integration2/schema/foreign-keys.spec.js#L65 ? To check that we are generating correct SQL statements.

Reply:
I notice the foreign key test only checks postgres and sqlite SQL generation. Is there tests elsewhere that checks generic SQL generation, or should I add an assertion for the other providers (MySQL, Oracle, MSSQL etc...)?

Reply:
sqlite would be plenty, postgre a nice bonus, but optional, everything else an overkill :)